language-support/hs/bindings/src/DA/Ledger/Valuable.hs

```diff
@@ -0,0 +1,40 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE OverloadedStrings #-}
+
+module DA.Ledger.Valuable(
+    Valuable(..), -- types which can be converted to/from a Ledger API Value
+    ) where
+
+import DA.Ledger.Types
+
+class Valuable a where
+    toValue :: a -> Value
+    fromValue :: Value -> Maybe a
+
+    toRecord :: a -> Record
+    toRecord =
+        Record Nothing
+        . map (\v -> RecordField {label = "", value = v})
+        . (\case VList vs -> vs; v -> [v])
+        . toValue
+
+    fromRecord :: Record -> Maybe a
+    fromRecord =
+        fromValue
+        . VList
+        . map (\RecordField{value} -> value)
+        . fields
+
+instance Valuable Int where
+    toValue = VInt
+    fromValue = \case VInt x -> Just x; _ -> Nothing
+
+instance Valuable Party where
+    toValue = VParty
+    fromValue = \case VParty x -> Just x; _ -> Nothing
+
+instance Valuable a => Valuable [a] where
+    toValue = VList . map toValue
+    fromValue = (mapM fromValue =<<) . \case VList vs -> Just vs; _ -> Nothing
```

Comment:
I find this hard to read. I'd have written something straightforward like
```haskell
fromValue = \case
  VList vs -> mapM fromValue v
  _ -> Nothing
```
I must admit, I'm not the biggest fan of the point free style in many cases.

Reply:
That's funny. Because I only introduced the =<< in response to a suggestion from HLint.
I also am not a big fan of point free in many cases.
Although I do find myself shifting in that direction:  Haskell infection perhaps? :)

Reply:
I originally wrote:
`fromValue = join . fmap (mapM fromValue) . (\case VList vs -> Just vs; _ -> Nothing)`
before HLint's suggestion. But clearly your suggestion is the simplest/best.

Reply:
I think HLint's suggestion makes your version better - a combo of join/fmap is very confusing, but it's still not as clear as Martin's.